quint/src/quintVerifier.ts

```diff
@@ -242,22 +191,33 @@ async function loadProtoDefViaReflection(): Promise<VerifyResult<ProtoPackageDef
   const serverReflectionService = reflectionProtoDescriptor.grpc.reflection.v1alpha.ServerReflection
   const reflectionClient = new serverReflectionService(APALACHE_SERVER_URI, grpc.credentials.createInsecure())
 
-  // Query reflection endpoint (retry if server is unreachable)
-  return retryWithTimeout(
-    () =>
-      new Promise<ServerReflectionResponse>((resolve, reject) => {
-        const call = reflectionClient.ServerReflectionInfo()
-        call.on('data', (r: ServerReflectionResponse) => {
-          call.end()
-          resolve(r)
-        })
-        call.on('error', (e: grpc.StatusObject) => reject(e))
+  // Wait for gRPC channel to come up, with 1sec pauses
+  if (retry) {
+    await (async () => {
+      for (;;) {
+        const grpcChannelState = reflectionClient.getChannel().getConnectivityState(true)
+        if (grpcChannelState == grpc.connectivityState.READY) {
+          return
+        } else {
+          await setTimeout(1000)
+        }
+      }
+    })()
```

Comment:
Why does this need to be a thunk if we are juts calling right away?

Reply:
Ah, good catch! I assume "a thunk" = "async"? 😄
Removed the async function wrapper in https://github.com/informalsystems/quint/commit/e78648f1b798148d335c3c22b5925260138d6c3b

Reply:
A "thunk" is a computation suspended with a nullary function: https://en.wikipedia.org/wiki/Thunk#Functional_programming

So here I refer to the fact that we have `(() => ...)()` rather than the `async`.